x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/EngineIndexService.java

```diff
@@ -308,7 +303,47 @@ public void deleteEngine(String engineName, ActionListener<DeleteResponse> liste
         } catch (Exception e) {
             listener.onFailure(e);
         }
+    }
+
+    GetAliasesResponse getAlias(String engineAliasName) {
+        return clientWithOrigin.admin().indices().getAliases(new GetAliasesRequest(engineAliasName)).actionGet();
+    }
+
+    private void removeAlias(String engineAliasName, ActionListener<AcknowledgedResponse> listener) {
+        IndicesAliasesRequest aliasesRequest = new IndicesAliasesRequest().addAliasAction(
+            IndicesAliasesRequest.AliasActions.remove().aliases(engineAliasName).indices("*")
+        );
+        clientWithOrigin.admin().indices().aliases(aliasesRequest, ActionListener.wrap(response -> {
+            if (response.isAcknowledged() == false) {
+                logger.warn("Request to remove alias [{}] response was not acknowledged", engineAliasName);
+            }
+            listener.onResponse(null);
+        }, listener::onFailure));
+    }
+
+    public void deleteEngineAndAlias(String engineName, ActionListener<DeleteResponse> listener) {
 
+        deleteEngine(engineName, new ActionListener<>() {
+            @Override
+            public void onResponse(DeleteResponse deleteResponse) {
+                removeAlias(Engine.getEngineAliasName(engineName), new ActionListener<>() {
+                    @Override
+                    public void onResponse(AcknowledgedResponse acknowledgedResponse) {
+                        listener.onResponse(null);
```

Comment:
Same here, I guess it doesn't hurt to provide the response to the `onResponse` method

Reply:
Good suggestion, changed!